osu.Game/Screens/Edit/List/DrawableList.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Edit.List
+{
+    public class DrawableList<T> : RearrangeableListContainer<T>, IDrawableListItem<T>
+        where T : Drawable
+    {
+        private Action onDragAction { get; set; }
+
+        public Action OnDragAction
+        {
+            get => onDragAction;
+            set
+            {
+                onDragAction = value;
+                UpdateItem();
+            }
+        }
+
+        private Action<SelectionState> selectAll;
+
+        public Action<SelectionState> SelectAll
+        {
+            get => selectAll;
+            set
+            {
+                selectAll = value;
+                UpdateItem();
+            }
+        }
+
+        private Func<T, LocalisableString> getName;
+
+        public Func<T, LocalisableString> GetName
+        {
+            get => getName;
+            set
+            {
+                getName = value;
+                UpdateItem();
+            }
+        }
+
+        public DrawableList()
+        {
+            getName = ((IDrawableListItem<T>)this).GetDefaultText;
+            selectAll = ((IDrawableListItem<T>)this).SelectableOnStateChanged;
+            onDragAction = () => { };
+
+            RelativeSizeAxes = Axes.X;
+            //todo: compute this somehow add runtime
+            Height = 100f;
+            ListContainer.Spacing = new Vector2(2.5f);
+            SelectAll = ((IDrawableListItem<T>)this).SelectableOnStateChanged;
+        }
+
+        protected override bool OnClick(ClickEvent e)
+        {
+            if (e.Button == MouseButton.Left)
+                SelectAll(SelectionState.NotSelected);
+            base.OnClick(e);
+            return e.Button == MouseButton.Left;
+        }
+
+        public void AddRange(IEnumerable<T>? drawables)
+        {
+            if (drawables is null) return;
+
+            foreach (T drawable in drawables)
+            {
+                addInternal(drawable);
+            }
+
+            OnDragAction();
+        }
```

Comment:
All this stuff seems too custom. I'd expect the list to just be a `Container` and handle things that way.